src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2336,7 +2339,24 @@ public static string ToBase64String(ReadOnlySpan<byte> bytes, Base64FormattingOp
             }
 
             bool insertLineBreaks = (options == Base64FormattingOptions.InsertLineBreaks);
-            string result = string.FastAllocateString(ToBase64_CalculateAndValidateOutputLength(bytes.Length, insertLineBreaks));
+            int outputLength = ToBase64_CalculateAndValidateOutputLength(bytes.Length, insertLineBreaks);
+
+            if (!insertLineBreaks && bytes.Length >= 64)
+            {
+                // For large inputs it's faster to allocate a temp buffer and call UTF8 version
+                // which is then extended to UTF8 via Latin1.GetString (base64 is always ASCI)
+                [MethodImpl(MethodImplOptions.NoInlining)]
+                static string ToBase64StringLargeInputs(ReadOnlySpan<byte> data, int outputLen)
+                {
+                    Span<byte> utf8buffer = outputLen <= 256 ? stackalloc byte[256] : new byte[outputLen];
+                    OperationStatus status = Base64.EncodeToUtf8(data, utf8buffer, out int _, out int _);
+                    Debug.Assert(status == OperationStatus.Done);
+                    return Encoding.Latin1.GetString(utf8buffer);
```

Comment:
Could be even faster when there's something like `Base64.EncodeToChar` which allows direct encoding into a char-buffer, so only one step is necessary instead of encoding to bytes, then bytes -> char.

Downside is more code to maintain, as the vectorized code-path in a `EncodeToChar` is a bit more complicated than just `(char)byteValue`.

Reply:
Yeah I just wanted to quickly optimize it for large inputs on both x64 and arm64. I had a much faster algorithm (and more friendly to smaller inputs) but it wasn't arm64 friendly